services/gps_nats/gps.py

```diff
@@ -32,15 +31,13 @@ async def run(loop):
     # Create nats object and connect to local nats server
     print('Setting up NATS')
     sys.stdout.flush()
-    nc = NATS()
-    await nc.connect("nats://localhost:4222")
-    #await nc.connect("nats://nats:4222")
+    nc = await nats.connect("localhost")
 
     for new_data in gps_socket:
         if new_data:
             fix = json.loads(new_data)
-            subject = "gps." + str(fix["class"])
-            print("Publishing new data point to subject", subject, ": ", new_data[:-1])
+            subject = os.getenv('HOSTNAME') + ".gps." + str(fix["class"])
+            #print("Publishing new data point to subject", subject, ": ", new_data[:-1])
```

Comment:
Maybe PREFIX or AVENA_PREFIX. In the multiple user/application case we would need more than one per host.

Small, but the first string join could be factored out of the hot loop.